src/genet/core.py

```diff
@@ -774,7 +774,8 @@ def empty_modes(mode_attrib):
 
         df["modes"] = df["modes"].apply(lambda x: persistence.setify(x))
 
-        df = df.loc[df.index.intersection(links)][df["modes"].apply(lambda x: bool(mode & x))]
+        df = df.loc[df.index.intersection(links)]
+        df = df[df["modes"].apply(lambda x: bool(mode & x))]
```

Comment:
I think this one had bug potential, those operations should not have been chained this way
The warning:
```python
/Users/kasia.kozlowska/PycharmProjects/CML/genet/src/genet/core.py:777: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df = df.loc[df.index.intersection(links)][df["modes"].apply(lambda x: bool(mode & x))]
```